clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/provider/artifacts/helm/HelmArtifactProvider.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Mirantis, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.provider.artifacts.helm;
+
+import com.netflix.spinnaker.cats.agent.Agent;
+import com.netflix.spinnaker.cats.agent.AgentSchedulerAware;
+import com.netflix.spinnaker.clouddriver.cache.SearchableProvider;
+import com.netflix.spinnaker.clouddriver.provider.ArtifactCloudProvider;
+import com.netflix.spinnaker.clouddriver.provider.cache.Keys;
+
+import java.util.*;
+
+public class HelmArtifactProvider extends AgentSchedulerAware implements SearchableProvider {
```

Comment:
Caching these is a little heavy-handed, since this will generate a lot of requests for a largely unchanging datastore.

What we found when caching docker images was that we quickly hit rate limits caching large numbers of images, when in reality the big value-add came from populating image drop-downs. 

If the work done by this caching agent can be switched to a live call against the helm repo for the sake of reducing request volume I think this would be in a much better state.